# 익스트림 스우 in 100에 포함되지 않는 직업이 존재할 확률은 1이다

> "작성 중"

- toc: true
- branch: master
- badges: true
- comments: true
- hide: true
- author: 한재수
- categories: [Statistics]

## 풀이에 오류가 존재하는 버전

In [1]:
import math

모든 직업에게 공평한 상황이라 익스우 in100에 포함될 확률이 동일하다고 가정하자

익스우 in100에 포함되지 않은 직업이 적어도 하나 이상 존재할 사건을 A라 하자

A의 여집합은 익스우 in100에 모든 직업이 포함되는 것이다

이때 P(A) = 1 - P(A^c)

P(A^c)를 계산하고 1에서 빼면 된다

직업별로 적어도 한 명 이상 포함되야 하므로 직업별로 1명씩 총 46명을 뽑고 나머지 54명은 남은 414명에서 아무나 뽑으면 된다

즉 분모는 460명 중 100명을 선택하는 사건의 개수이며 분자는 414명 중 54명을 선택하는 사건의 개수이다

라고 생각했는데 아니었다

왜냐하면 같은 직업 내 사람은 동일한 원소로 취급되기 때문에 위와 같이 계산하면 과대평가 된다

그래서 단순히 조합을 사용하면 안된다

In [2]:
x = math.comb(460, 100)
y = math.comb(414, 54)
answer = y / x
print(1 - answer)

1.0


단순히 460명에서 100명을 뽑는 것이 아니다

직업 46개 별로 각 인원 10명은 동일한 원소로 취급된다.... (중복 조합)

그런데 무지성으로 중복조합을 할 수도 없는데 왜냐하면 각 직업 인원에 10명이라는 제한이 걸려있기 때문이다

P(A)를 줄일려면 전체 도전 인원수가 줄어야 한다

위에서 k를 10으로 했는데 5로 줄여보자

즉, 46 * 5 = 230명이 익스우 in100에 도전하는 상황으로 바꾸자

In [4]:
x = math.comb(230, 100)
y = math.comb(230 - 46, 54)
answer = y / x
print(1 - answer)

1.0


k를 가장 작게 한다면 3이 된다

1과 2는 의미가 없는게 전체 도전 인원 수가 100보다 작아져서 모든 인원이 익스우in100에 성공한다

이제 k를 3으로 가정해보자

즉, 46 * 3 = 138명이 익스우 in100에 도전하는 상황으로 바꾸자

In [5]:
x = math.comb(138, 100)
y = math.comb(138 - 46, 54)
answer = y / x
print(1 - answer)

0.9999999927308524


In [6]:
# k가 3인 경우 드디어 P(A)가 정확히 1은 아니고 거의 1이다!! 우와 !!!!!

In [7]:
# 참고로 익스우 in100에 도전하는 전체 인원 수를 M이라 했을 때
# M을 전체 직업에 균등하게 분배해야 P(A)가 가장 작다 (정확히는 엔트로피가 가장 클 때 P(A)는 가장 작음)

In [9]:
# 결론: 모든 직업이 공평한 상황일지라도 익스우 in100에 포함되지 않는 직업이 존재할 확률은 1이다

## 풀이에 오류를 없앴다고 생각하는 버전

`-` 오류를 없애기 위해 k를 100으로 가정하자

`-` 그러면 한계치가 커졌기에 무지성으로 중복 조합을 고려해도 된다

`-` 최악의 경우를 가정하면 한 직업에서 모조리 100명을 뽑는 것인데 각 직업별 인원수가 100명이므로 실현 가능하다 (k가 이보다 작으면 불가능하다) 

In [2]:
46000-46

45954

`-` 각 직업별로 100명씩 총 46000명이 익스우 in 100에 도전한다고 가정하자

`-` 직업별로 적어도 한 명 이상 포함되야 하므로 직업별로 1명씩 총 46명을 뽑고 나머지 54명은 남은 45954명에서 아무나 뽑으면 된다

`-` 분모는 46개 직업 중 중복을 허용하여 100명을 선택하는 사건의 개수이며 분자는 46개 직업 중 중복을 허용하여 54명을 선택하는 사건의 개수이다

`-` 참고로 n개의 원소 중 중복을 고려하여 r개를 뽑는 사건의 개수는 n + r - 1개의 원소 중 r개를 순서를 고려하지 않고 선택하는 사건의 개수와 동일하다

In [3]:
x = math.comb(46 + 100 - 1, 100)
y = math.comb(46 + 54 - 1, 54)
answer = y / x
print(1 - answer)

0.9999999995311809


`-` 확률은 거의 1이다

`-` 결론은 오류가 있는 버전과 동일하다

`-` 그런데 위와 같이 계산하게 되면 각 직업별로 인원 수가 동일하면서 100명 이상이기만 하면 P(A)는 동일하게 계산된다

`-` 즉 전체 인원 수와 P(A)는 상관이 없다 (전체 인워 수가 커진다고 P(A)가 커지지 않음)